### ISSUE TITLE
New undetected data from ComfyUI generated image

### ISSUE THREAD

**Issue by ermanitu** [zanllp/sd-webui-infinite-image-browsing#756]
I am aware that it is very difficult to extract the data from any image generated by ComfyUI, but I have detected another image in which it does not read them correctly. If you find it useful, I have the python code to extract the prompt from JSON from the workflow that goes inside the image. Basically you have to go through the nodes until you find the CLIPTextEncode or CLIPTextEncodeFlux type and, in case there are several, it is usual that the one with the smallest id is the positive prompt.

I also attach the python code with which I extract these contents from JSON.

I love your application. Thank you.

`
    def get_prompt (self): # get the prompts in the workflow
        for node in self.workflow.values():
            if node["class_type"] == "CLIPTextEncode": # sd style
                if 'inputs' in node and 'text' in node['inputs']:
                    print(node['inputs']['text'])
            elif node["class_type"] == "CLIPTextEncodeFlux": # flux style
                if 'inputs' in node and 't5xxl' in node['inputs']:
                    print(node['inputs']['t5xxl'])`
`
![Imagen PNG-C27F9B89F6FB-1](https://github.com/user-attachments/assets/7f78f0cf-1e52-4dbb-a31b-468207b5f8b8)

**Comment by zanllp** [comment]
Update  to the latest IIB

**Comment by ermanitu** [comment]
Sorry, problem persist.

**Comment by ermanitu** [comment]
No, is it's my own program using the comfyUI API. I think the image returned by the API is missing some data. ComfyUI open those images without problems. I will investigate. Thanks.

**Comment by zanllp** [comment]
<img width="755" alt="image" src="https://github.com/user-attachments/assets/d52d19b1-0d33-4b9c-9185-03ed1c671d39">

**Comment by ermanitu** [comment]
Could you send me the details of 'prompt' and 'workflow' values?
I am guessing that the problem can be the order of the nodes. I will generate another image ordering the nodes and test in IB.

**Comment by ermanitu** [comment]
I sorted the JSON keys but the problem persists. I think you must use the value of 'prompt', not the value of 'workflow'.

**Comment by zanllp** [comment]
'workflow' is only used to determine whether the image was generated by ComfyUI; in fact, it is still 'prompt' during parsing.



https://github.com/zanllp/sd-webui-infinite-image-browsing/blob/1a0f07125845609dfe781492d68b34b81119f47f/scripts/iib/tool.py#L396-L403

**Comment by ermanitu** [comment]
Ok, I will try then embeed a workflow in the image to resolve the problem. Thanks!

**Comment by ermanitu** [comment]
Ops! Not possible. ComfyUI API requires saving the workflows in different format (saved by "save API format") than the format used by the ComfyUI itself. The only option is checking ComfyUI generation not using the 'workflow' attribute. It's possible to infer ComfyUI by some property of the 'prompt' variable?

**Comment by zanllp** [comment]
I'll check how to implement it later.
 Of course, if you have any ideas, feel free to propose a PR!
